src/main/java/com/mongodb/hibernate/dialect/MongoDialectSettings.java

```diff
@@ -0,0 +1,293 @@
+ *             if {@linkplain ConnectionString#getDatabase() configured};
+ *             otherwise there is no default value, and a value must be configured via {@link Builder#databaseName(String)}.</td>
+ *         </tr>
+ *         <tr>
+ *             <td>{@link #isAutoCommit()}</td>
+ *             <td>✓</td>
+ *             <td>{@value AvailableSettings#AUTOCOMMIT}</td>
+ *             <td>
+ *                 <ul>
+ *                     <li>{@link String}, only {@code "true"}, {@code "false"} are allowed</li>
+ *                     <li>{@link Boolean}</li>
+ *                 </ul>
+ *             </td>
+ *             <td>{@value AvailableSettings#AUTOCOMMIT}, if configured; otherwise {@code false}.</td>
+ *         </tr>
+ *     </tbody>
+ * </table>
+ *
+ * @see MongoDialectConfigurator
+ */
+public final class MongoDialectSettings {
+    private final MongoClientSettings mongoClientSettings;
+    private final String databaseName;
+    private final boolean autoCommit;
+
+    private MongoDialectSettings(Builder builder) {
+        this.mongoClientSettings = builder.mongoClientSettingsBuilder.build();
+        this.databaseName = notNull("databaseName", builder.databaseName);
+        this.autoCommit = builder.autoCommit;
+    }
+
+    /**
+     * Gets the {@link MongoClientSettings} to be used when creating the internal {@link MongoClient}.
+     *
+     * @return The {@link MongoClientSettings}.
+     * @see Builder#applyToMongoClientSettings(Consumer)
+     */
+    public MongoClientSettings getMongoClientSettings() {
+        return mongoClientSettings;
+    }
+
+    /**
+     * Gets the name of a MongoDB database used as the {@linkplain Connection#getSchema() JDBC schema} of a
+     * {@linkplain Connection connection} {@linkplain MongoConnectionProvider#getConnection() obtained} from
+     * {@link MongoConnectionProvider}.
+     *
+     * @return The name of the default MongoDB database.
+     * @see Builder#databaseName(String)
+     * @see DatabaseMetaData#getSchemaTerm()
+     */
+    public String getDatabaseName() {
+        return databaseName;
+    }
+
+    /**
+     * The {@linkplain Connection#getAutoCommit() auto-commit mode} of {@linkplain Connection connections}
+     * {@linkplain MongoConnectionProvider#getConnection() obtained} from {@link MongoConnectionProvider}.
+     *
+     * @return The default auto-commit mode.
+     * @see Builder#autoCommit(boolean)
+     */
+    public boolean isAutoCommit() {
+        return autoCommit;
+    }
+
+    /**
+     * Creates a new {@link MongoDialectSettings.Builder} based on {@code configProperties}.
+     *
+     * @param configProperties The {@linkplain Configurable#configure(Map) configuration properties}.
+     * @return A new {@link MongoDialectSettings.Builder}.
+     */
+    public static MongoDialectSettings.Builder builder(Map<String, Object> configProperties) {
+        return new MongoDialectSettings.Builder(notNull("configProperties", configProperties));
```

Comment:
I think given `configProperties` is not marked as `@Nullable`, we could save the usage of `notNull()` and trust `NullAway` to cover our asses (in theory at least any usage inconsistent with the `jspecify` annotation would be treated as bug during compiling phase). This is one of improvements in this project compared to Java driver project.